cirq/ops/common_channels.py

```diff
@@ -184,6 +195,700 @@ def depolarize(p: float) -> DepolarizingChannel:
             the Pauli gates is applied independently with probability p / 3.
 
     Raises:
-        ValueError if p is not a valid probability.
+        ValueError: if p is not a valid probability.
     """
     return DepolarizingChannel(p)
+
+
+class GeneralizedAmplitudeDampingChannel(raw_types.Gate):
+    """Dampen qubit amplitudes through non ideal dissipation.
+
+    This channel models the effect of energy dissipation to the
+    surrounding environment at some particular rate, with some
+    given probability at a non zero temperature.
+    """
+
+    def __init__(self, p, gamma) -> None:
+        """The generalized amplitude damping channel.
+
+        Construct a channel to model the energy dissipation to the
+        environment at rate gamma, occuring with probability p
+
+        This channel evolves a density matrix via
+            \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
```

Comment:
maybe better to just write this as 
\rho -> \sum_{i=0}^3 M_i \rho M_i^\dagger
?  Not sure either way is fine.

Reply:
I kept it the way it it is , since this new form is pretty similar to what is already in protocols.